### PR TITLE
new bottom nav design

### PR DESCRIPTION
**Pull request by rasom**
screen shot 2019-02-22 at 12 03 29

### CODE REVIEW

**Comment by rasom** [comment]
@errorists I'm trying to understand if the shadow looks good enough on iOS, would be great to get some feedback

**Comment by rasom** [comment]
@flexsurfer @yenda @jeluard @goranjovic 
Hey guys, do we have some kind of shadows implemented for Android anywhere in the app?

**Comment by flexsurfer** [comment]
@rasom i believe we don't , we used elevation

**Comment by rasom** [comment]
@flexsurfer any specific place where I can find how it looks?

**Comment by errorists** [comment]
@rasom hey man, looks great! 👏

**Comment by flexsurfer** [comment]
@rasom yes i believe toolbar on the home screen

**Comment by errorists** [comment]
also, if you need to use elevation on Android, the correct value is 8 for the bottom bar.

**Comment by rasom** [comment]
It didn't work when I first tried it because elevation works only if background color specified... 
So that's how it looks on Android with elevation 8 

photo_2019-02-25_10-36-01

**Comment by errorists** [comment]
🤩 looking good, 8 it is then. wasn't sure myself since elevation is proprietary to android and doesn't translate to usual shadow values, 8 is what Google specifies in Material for bottom bars.

File: src/status_im/ui/components/bottom_bar/core.cljs [hunk 1]
```diff
@@ -66,25 +67,84 @@
...
+    [new-tab {:icon    :main-icons/message
+              :label   "Chats"
+              :active? (= current-view-id :chat-stack)}]
+    [new-tab {:icon    :main-icons/dapp
+              :label   "DApps"
+              :active? (= current-view-id :dapp-stack)}]
+    [new-tab {:icon    :main-icons/wallet
+              :label   "Wallet"
+              :active? (= current-view-id :wallet-stack)}]
+    [new-tab {:icon    :main-icons/profile
```
Comment (errorists) [inline src/status_im/ui/components/bottom_bar/core.cljs]: hey @rasom could we correct the icon for profile as part of this PR, it should be using the `user_profile` and not `profile` 🙏

File: src/status_im/ui/components/bottom_bar/core.cljs [hunk 2]
```diff
@@ -66,25 +67,84 @@
...
+   [react/view {:style {:height         height
+                        :align-self     :stretch
+                        :padding-left    8
+                        :padding-right   8
+                        :flex-direction :row}}
+    [new-tab {:icon    :main-icons/message
+              :label   "Chats"
+              :active? (= current-view-id :chat-stack)}]
+    [new-tab {:icon    :main-icons/dapp
+              :label   "DApps"
```
Comment (errorists) [inline src/status_im/ui/components/bottom_bar/core.cljs]: hey @rachelhamlin should the tab say `DApps` or `ÐApps` or even something entirely else like 'Browser'?

**Comment by rasom** [comment]
@errorists do we have a new design for message counter?

photo_2019-02-26_15-04-19

File: src/status_im/ui/components/bottom_bar/core.cljs (hunk 1 above)
Comment (rasom) [reply src/status_im/ui/components/bottom_bar/core.cljs]: sure

**Comment by errorists** [comment]
@rasom no, we didn't update the badge design in any way, at one point we thought about removing the numbers from tab bar badges but didn't pursue that any further. From the Figma components list we're using the [Small Badge](https://www.figma.com/file/cb4p8AxLtTF3q1L6JYDnKN15/Index?node-id=1213%3A4l)

**Comment by rasom** [comment]
@errorists the link is broken

**Comment by errorists** [comment]
@rasom sorry for that, corrected.

File: src/status_im/ui/components/bottom_bar/core.cljs (hunk 2 above)
Comment (rachelhamlin) [reply src/status_im/ui/components/bottom_bar/core.cljs]: For now, I tend to think leave it as `ÐApps`.

**Comment by rasom** [comment]
that's how it looks on iOS now

**Comment by errorists** [comment]
it's awesome, one thing I'd suggest if that's not too much work is to align the horizontal center of the badge with the right edge of the icon, alternatively nudge it towards right by 4px, so it doesn't cover the icon as much. My [best attempt at Figma-fying it ](https://www.figma.com/file/deXIBZm6fU1CCL0voYbfTujz/Navigation?node-id=234%3A1302)

**Comment by rasom** [comment]
@errorists

**Comment by errorists** [comment]
perfect, no more questions from me 👍

**Comment by rachelhamlin** [comment]
Looks awesome @rasom. At this rate we'll probably have new navigation ready on nightlies by the time the new site goes live in a few weeks?

**Comment by rasom** [comment]
@rachelhamlin yes, most likely. I will wrap this particular PR today and it will be reviewed/merged today or tomorrow. I don't think that it will take too much time. The next one is animation on switching to chat (bottom naw moves down a bit) and then probably some adjustments to "create chat" flow etc.

@rachelhamlin @errorists, the only question I have about this PR: we still haven't restructured navigation stacks for a new DApps tab, so it probably will be hidden till it happens. Is that ok?

**Comment by rachelhamlin** [comment]
@rasom absolutely. I realized I created this issue a few days back to track all changes - https://github.com/status-im/status-react/issues/7534

**Review by errorists — APPROVED** [review]
Looks beautiful here on iOS, thanks Roman 🙏  An unrelated comment, I have a feeling there's a performance downgrade in react-navigation transitions, especially when opening chats but it's subjective and have nothing to prove it. I'd just love if we could someday figure this stuff out.

**Comment by rasom** [comment]
@errorists I heard the same from @mandrigin today, but I doubt that he run this PR, so it's probably related to a different change. Right now I don't know what's the reason...

**Comment by rasom** [comment]
@errorists is it only chats->chat or other transitions too?

**Comment by errorists** [comment]
thanks, glad to hear I'm not imagining things. the only thing that comes to my mind is the searchbar PR, I migrated from a ~yesterday's nightly that didn't have it and I saw the difference compared to that nightly build. To answer, it's there in all transitions but it's most noticeable in chats because the content gets rendered there when the transition happens. I'll try to see if it's visible in a 60fps video capture

**Review by yenda — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

**Review by vitvly — APPROVED** [review]

**Comment by churik** [comment]
@rasom 
on latest builds https://i.diawi.com/tmjMbF and https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190227-091852-67fbee-pr.apk I don't see DApps in bottom navigation and instead of "Chats" I see  "Home".
img_cb481b5629bd-1
Can you please take a look?

<blockquote><div><strong><a href="https://i.diawi.com/tmjMbF">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote></blockquote>

**Comment by rasom** [comment]
@churik see https://github.com/status-im/status-react/pull/7550#issuecomment-467772299 regarding dapps, will fix `"home"` thing in a moment

**Comment by rasom** [comment]
@churik fixed

**Comment by churik** [comment]
@rasom one issue found on Android 8.0. When you create wallet from chat or chat from Dapp -> bottom navigation is inaccessible (so if you tap nothing happens) and it is cut.

_Scenario 1:_ 
- open any public chat
- tap on ...
- tap on Wallet 


_Scenario 2:_
- open any DApp
- tap on Chat icon near reload button


And I have a general question about PR - is it only about changes in design in the bottom navigation bar or it contains some more changes in navigation?

**Comment by rasom** [comment]
@churik actually these two routes are going to be removed. I can remove modal `:wallet` in chat right now (it doesn't make sense as you can switch to the wallet tab), though the second case requires Dapps tab to be enabled and this will happen soon.

**Comment by rasom** [comment]
@churik btw is that android 8.0 or rather galaxy s8/9 specific issue?

**Comment by churik** [comment]
@rasom I'm using LG V20 phone, so definitely not Galaxy

**Comment by rasom** [comment]
ok I see
@churik I would merge it with this issue and remove this route later when DApp tab will be enabled. We can create an issue and track it, but still I'm not sure it is worth efforts at the moment

**Comment by churik** [comment]
Mentioned in https://github.com/status-im/status-react/issues/7534#issuecomment-468605503,  should be enough

**Pull request merged**